pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -297,81 +319,21 @@ func (r *ReconcileOneAgent) reconcileRollout(logger logr.Logger, instance dynatr
 	return updateCR, nil
 }
 
-func (r *ReconcileOneAgent) determineOneAgentPhase(instance dynatracev1alpha1.BaseOneAgentDaemonSet) (bool, error) {
-	var phaseChanged bool
-	dsActual := &appsv1.DaemonSet{}
-	err := r.client.Get(context.TODO(), types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}, dsActual)
-
-	if k8serrors.IsNotFound(err) {
-		return false, nil
-	}
-
-	if err != nil {
-		phaseChanged = instance.GetOneAgentStatus().Phase != dynatracev1alpha1.Error
-		instance.GetOneAgentStatus().Phase = dynatracev1alpha1.Error
-		return phaseChanged, err
-	}
-
-	if dsActual.Status.NumberReady == dsActual.Status.CurrentNumberScheduled {
-		phaseChanged = instance.GetOneAgentStatus().Phase != dynatracev1alpha1.Running
-		instance.GetOneAgentStatus().Phase = dynatracev1alpha1.Running
-	} else {
-		phaseChanged = instance.GetOneAgentStatus().Phase != dynatracev1alpha1.Deploying
-		instance.GetOneAgentStatus().Phase = dynatracev1alpha1.Deploying
-	}
-
-	return phaseChanged, nil
-}
-
-func (r *ReconcileOneAgent) reconcileVersion(logger logr.Logger, instance dynatracev1alpha1.BaseOneAgentDaemonSet, dtc dtclient.Client) (bool, error) {
-	updateCR := false
-
-	// get desired version
-	desired, err := dtc.GetLatestAgentVersion(dtclient.OsUnix, dtclient.InstallerTypeDefault)
-	if err != nil {
-		return false, fmt.Errorf("failed to get desired version: %w", err)
-	} else if desired != "" && instance.GetOneAgentStatus().Version != desired {
-		logger.Info("new version available", "actual", instance.GetOneAgentStatus().Version, "desired", desired)
-		instance.GetOneAgentStatus().Version = desired
-		updateCR = true
+func (r *ReconcileOneAgent) ReconcilePullSecret(instance dynatracev1alpha1.BaseOneAgent, log logr.Logger) error {
```

Comment:
Does `ReconcilePullSecret` have to be public?

Reply:
Changed to private